portal-service/src/com/liferay/portal/model/PortletInstance.java

```diff
@@ -179,4 +240,45 @@ private void validatePortletName(String portletName) {
 	private final String _portletName;
 	private final long _userId;
 
+	private static final class FullInstanceId {
```

Comment:
I like the idea of having a class, but it cannot be called exactly the same as the name of the variable. You need to pick a different name for the class.